### DISCUSSION TITLE
[Nuxt3] Use layouts with `<script setup>`

### DISCUSSION THREAD

**Discussion by mrleblanc101** [nuxt/nuxt#18102] [Questions] [answered]
The doc is not very clear. https://v3.nuxtjs.org/docs/directory-structure/layouts
I'd think I'd have to use the new `<NuxtLayout>` component, but the it says you have to disable something using the option API.
```
<script>
export default {
  layout: false,
};
</script>
```

**Comment by lucpotage** [comment]
From what I understand, you only need to disable the layout property when using the `<NuxtLayout>` component that enables more advanced features around slots.

If you need simple layouts with one default slot like in Nuxt 2, there is no change, except changing `<Nuxt/>` with `<slot/>` component.

**Comment by ckam99** [comment]
Maybe he want do something like this: 
![download (1)](https://user-images.githubusercontent.com/72167620/137399502-4c98f9ed-3013-47f5-a0eb-0f0b03a80d65.png)

As Vue 3.2 with `<script setup>` option does not allow to export module,  there is an alternative option to do something like ?
```vue
<NuxtLayout name="custom" layout="false">
...
</NuxtLayout>
```

instead to :
```vue
<template>
  <NuxtLayout name="custom">
    <template #header> Some header template content. </template>

    The rest of the page
  </NuxtLayout>
</template>

<script>
export default {
  layout: false,
};
</script>
```

**Comment by wobsoriano** [comment]
You can disable layout and have `script setup`.

```html
<script setup lang="ts">
const nuxtApp = useNuxtApp()
console.log(nuxtApp)
</script>

<template>
  <NuxtLayout name="custom">
    <template #header> Some header template content. </template>

    The rest of the page
  </NuxtLayout>
</template>

<script lang="ts">
export default {
    layout: false
}
</script>
```

`<script>` and `<script setup>` must have the same language type.

More info: https://v3.vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script

**Reply by ckam99** [reply]
Thank you. It works for me

**Reply by wobsoriano** [reply]
Can you mark this as answered?

**Comment by crypto-magic** [comment, marked as answer]
https://v3.vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script

`<script setup>` can be used alongside normal `<script>`

```vue
<script>
export default {
  layout: false,
}
</script>

<script setup>
// executed in setup() scope (for each instance)
</script>
```

**Comment by kaemu** [comment]
`/layouts/main.vue`

```html
<template>
  <div>
    Navigation…
    <slot />
    Footer…
  </div>
</template>
```

`/pages/index.vue`

```html
<template>
  <NuxtLayout name="main">
    My page content…
  </NuxtLayout>
</template>

<script setup lang="ts">
</script>
```

No need additional code
👉Think to restart `npm run dev` after you first created the layout file

**Reply by mrleblanc101** [reply]
This is wrong... look at the docs

**Reply by kaemu** [reply]
What is wrong ? 🧐 
This works for me

**Reply by mrleblanc101** [reply]
Did you not read anything about the thread ? Including the link in my first post ?

**Comment by aussieboi** [comment]
Hi, to continue on the layout discussion and not to open a new discussion, want to add a suggestion, as people see benefits of  `<script setup> ` and using it more, myself including, just food for thought maybe similar functionality with `<template>` block 

following would be resolved to  `default` layout.
```
<template>
</template>
```
or
```
<template layout="default">
</template>
```
or even 

```
<template default>
</template>
```
personally, this would be too big an abstraction for newcomers, and maybe the learning curve would suffer too.

Any other value in the `layout` attribute would resolve to the same-named `.vue` file in the `layouts/` folder.
This would avoid following an additional `script` block where we would only need to specify layout.

```
<script>
export default {
  layout: false,
}
</script>
```

Let me know what you think. 

Thanks

**Comment by tcampbPPU** [comment]
Not having to add 2 scripts in file would be better imo as describe in the version 3 [docs layouts](http://v3.nuxtjs.org/docs/directory-structure/layouts)

This seems more elegant to me:
```vue
<script lang="ts" setup>
import { useLayout } from '#app'

useLayout('custom')
</script>
```

**Reply by acidjazz** [reply]
I like this option, also 

```html
<template layout="custom">
```

and then no layout="" means `layouts/default.vue`
